### PR TITLE
Add File::Stat

### PR DESCRIPTION
**Pull request by nanobowers**
Added the `File::Stat` class, and an implementation of `File::Stat#ftype` that is just functional enough so that the `spec/core/file/stat/new_spec.rb` test would pass.

I am not sure if I went about this the best way, I took a look at the `SPECIAL_CLASSES_WITHOUT_DEDICATED_TYPES` array in binding_gen.rb but didnt grok how I could use that to my advantage.

### CODE REVIEW

**Comment by nanobowers** [comment]
I must have broken something when i tried to rebase, ugh. 😞

File: include/natalie/file_stat_object.hpp [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+#include "natalie/forward.hpp"
+#include "natalie/object.hpp"
+#include <filesystem>
+
+namespace Natalie {
+
+class FileStatObject : public Object {
+public:
+    FileStatObject(const StringObject &other)
```
Comment (seven1m) [inline include/natalie/file_stat_object.hpp:11]: This argument is unused, yes?

File: src/file_object.cpp [hunk 2]
```diff
@@ -404,4 +404,9 @@ Value FileObject::chmod(Env *env, Value mode, Value path) {
     return Value::integer(1); // return # of files
 }
 
+Value FileObject::stat(Env *env, Value path) {
+    path = fileutil::convert_using_to_path(env, path);
+    return new FileStatObject { path->as_string()->c_str() };
```
Comment (nanobowers) [inline src/file_object.cpp:409]: It gets used here, though it is kind of stubbed out at this point.

File: include/natalie/file_stat_object.hpp (hunk 1 above)
Comment (nanobowers) [reply include/natalie/file_stat_object.hpp:11]: Unsure how to point to a line of code, but there is a comment below showing where the argument gets used in file_object.cpp line 409.  While it's stubbed out right now and the other argument goes unused (probably should be renamed from `other` to `path`, actually)... I do intend to use it it in the future for calls from `File.stat`, and `File#stat`, etc as opposed to `File::Stat.new`

File: src/file_object.cpp (hunk 2 above)
Comment (seven1m) [reply src/file_object.cpp:409]: Oh oK I'll wait till you're done. 😁

Comment (nanobowers) [reply src/file_object.cpp:409]: This is my first foray into adding a new class and was mainly looking for early feedback on this draft to let me know if I really did things the hard way here - before I got too far along - or any other advice you might have.  It's still really unpolished/unfinished.

**Comment by nanobowers** [comment]
This is currently gated on the macos compile and unfortunately I don't have access to a MacOS development environment.
In particular, the issue is that I'm including `<sys/sysmacros.h>` in linux, presumably there is an equivalent way to include the `major()` and `minor()` functions or defined macros.  If someone could offer advice or help on this, would be appreciated.

**Pull request merged**